tests/test_integrated_channels/test_integrated_channel/test_transmitters/test_learner_data.py

```diff
@@ -38,6 +39,23 @@ def setUp(self):
 
         self.learner_transmitter = LearnerTransmitter(self.enterprise_config)
 
+    def test_transmit_single_learner_data_signal_kwargs(self):
+        """ transmit_single_learner_data is called with kwargs as a shared task from OpenEdx,
+        so test that interface hasn't changed. """
+
+        edx_platform_api_signal_kwargs = {
+            'username': "fake_username",
+            'course_run_id': "TEST_COURSE_RUN_KEY"
+        }
+        # If you change the arg names to transmit_single_learner_data
+        # this test will now fail with a TypeError:
+        try:
+            transmit_single_learner_data(**edx_platform_api_signal_kwargs)
+        except TypeError:
+            assert False  # if we see a Type error, the test should fail because we failed with bad keyword parameters
+        except Exception:   # pylint: disable=broad-except
```

Comment:
This feels like a weird pass condition. I'm also generally not a fan of highly specific tests that only really avoid 1 scenario we've already hit. Ideally I'd rather us actually test the logic of this method but yeah that's probably nightmarish.

Reply:
Oh this does not test the function at all: the problem is that unlike for REST API violations, no test exists to catch if we violate the plugin interface with the wrong params.

Reply:
You could do something like the converse: call it with the wrong keyword(s) and do a `with self.assertRaises(...):`

Reply:
That wouldn't have caught this issue though would it? The issue was that we were in fact passing the wrong keywords. Thus a test that asserts that error gets raised is saying "yes this fails because these keywords are wrong and we don't want them. That is correct and thus this test passes!" would also have passed before this code change?

Reply:
@taliaedX you are correct: the goal of the test was defensive coding that would raise an error if someone did what I did and changed a method signature. However, I do like @macdiesel 's solution going forward of updating the doc strings to note that the method signatures are actually an API contract instead, as that provides defensive coding right where the programmer can see it. I will make another small doc-only PR with a docstring note in anything in integrated channels we call as a kwarg.